### PR TITLE
Spec Rand

### PR DESCRIPTION
**Pull request by amin1377**
Add support to use SPEC random number generator if SPEC_CPU is defined.

@heshpdx 
@vaughnbetz

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -298,6 +299,16 @@ if(VTR_ENABLE_VERBOSE)
     message(STATUS "Enabling increased debugging verbosity")
 endif()
 
+#
+# Build for SPEC CPU Benchmark v8
+#
+set(SPEC_CPU_FLAGS "")
+if(SPEC_CPU)
+    #Enable gcov
```
Comment (vaughnbetz) [inline CMakeLists.txt:307]: Why does this say enable gcov?

Comment (vaughnbetz) [reply CMakeLists.txt:307]: Comment on what this define is for ... what has to change / what's the purpose?  (spec benchmarking version of vtr; maximizes portability of code and mimimizing the variation in how much work we do on different platforms).

File: libs/libvtrutil/CMakeLists.txt [hunk 2]
```diff
@@ -83,7 +83,7 @@ add_custom_command(OUTPUT ${VTR_VERSION_FILE_OUT}
 #
 # Source files and library
 #
-file(GLOB_RECURSE LIB_SOURCES src/*.cpp)
+file(GLOB_RECURSE LIB_SOURCES src/*.cpp src/*.c)
```
Comment (vaughnbetz) [inline libs/libvtrutil/CMakeLists.txt:86]: Probably easier to make the new specrand.c file into specrand.cpp --> ask Mahesh if that's OK.  If not, we can do this.

File: libs/libvtrutil/src/specrand.c [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+   NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+   SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+   Any feedback is very welcome.
+   http://www.math.sci.hiroshima-u.ac.jp/~m-mat/MT/emt.html
+   email: m-mat @ math.sci.hiroshima-u.ac.jp (remove space)
+*/
+/* Slightly modified for use in SPEC CPU by Cloyce D. Spradling (5 Nov 2009)
+ */
```
Comment (vaughnbetz) [inline libs/libvtrutil/src/specrand.c:46]: Comment why this is in vtr --> for inclusion in the SPEC cpu benchmarks, and it helps ensure we get the random numbers on different platforms.

**Review by vaughnbetz — COMMENTED** [review]
A few commenting suggestions.

File: libs/libvtrutil/CMakeLists.txt (hunk 2 above)
Comment (heshpdx) [reply libs/libvtrutil/CMakeLists.txt:86]: It's up you how you want to pull in these files. If you'd like to convert to cpp, please go ahead. There is no requirement from my side :-)

**Pull request merged**